### PR TITLE
Adding Azure SQL Database templates

### PR DESCRIPTION
**Pull request by croblesm**
## What type of PR is this?

- [X] Add a new dev container collection
- [ ] Update to an existing dev container collection
- [ ] Documentation/spec update
- [ ] Other containers.dev site update (UX, layout, etc)

## Description

Hello there 👋

With this PR, I'm updating collection-index.yml file to include the Azure SQL Database templates. The main repo can be found [here](https://github.com/microsoft/azuresql-devcontainers).

I have been working with @samruddhikhandale  and @bamurtaugh on this project.

### Collection checklist
_If your PR contributes a new collection, please utilize this checklist:_
- [X] Collection name
- [X] Maintainer name
- [X] Maintainer contact link (i.e. link to a GitHub repo, email)
- [X] Repository URL
- [X] OCI Reference
- [X] I acknowledge that this collection provides new functionality, distinct from the existing collections part of this index.

### CODE REVIEW

**Review by samruddhikhandale — APPROVED** [review]
Looks good, thanks!

File: _data/collection-index.yml [hunk 1]
```diff
@@ -8,6 +8,11 @@
   contact: https://github.com/devcontainers/templates/issues
   repository: https://github.com/devcontainers/templates
   ociReference: ghcr.io/devcontainers/templates
+- name: Azure SQL Database Dev Container Templates
+  maintainer: Azure SQL Developer Experiences Team
+  contact: https://github.com/microsoft/azuresql-devcontainers/issues
+  repository: https://github.com/microsoft/azuresql-devcontainers
+  ociReference: ghcr.io/microsoft/azuresql-devcontainers
```
Comment (eitsupi) [inline _data/collection-index.yml:15]: Just a question: why was this added in this position instead of at the end?

Comment (eitsupi) [reply _data/collection-index.yml:15]: @samruddhikhandale Any thoughts?
